example/cajita_tutorial/11_interpolation/interpolation_example.cpp

```diff
@@ -0,0 +1,224 @@
+    Kokkos::parallel_for(
+        "fill_points", createExecutionPolicy( cell_space, ExecutionSpace() ),
+        KOKKOS_LAMBDA( const int i, const int j ) {
+            int pi = i - halo_width;
+            int pj = j - halo_width;
+            int pid = pi + cell_space.extent( Cajita::Dim::I ) * pj;
+            int idx[2] = { i, j };
+            double x[2];
+            local_mesh.coordinates( Cajita::Cell(), idx, x );
+            points( pid, Cajita::Dim::I ) = x[Cajita::Dim::I];
+            points( pid, Cajita::Dim::J ) = x[Cajita::Dim::J];
+        } );
+
+    // Next, we use Cajita functionality to create grid data fields.
+    // Create a scalar field on the grid. See the Array tutorial example for
+    // information on these functions.
+    auto scalar_layout =
+        Cajita::createArrayLayout( local_grid, 1, Cajita::Node() );
+    auto scalar_grid_field = Cajita::createArray<double, ExecutionSpace>(
+        "scalar_grid_field", scalar_layout );
+
+    // Create a halo for scatter operations. This concept is discussed in more
+    // detail in the Halo tutorial example.
+    auto scalar_halo =
+        Cajita::createHalo( *scalar_grid_field, Cajita::NodeHaloPattern<2>() );
+    auto scalar_grid_host =
+        Kokkos::create_mirror_view( scalar_grid_field->view() );
+
+    // Create a vector field on the grid.
+    auto vector_layout =
+        Cajita::createArrayLayout( local_grid, 2, Cajita::Node() );
+    auto vector_grid_field = Cajita::createArray<double, ExecutionSpace>(
+        "vector_grid_field", vector_layout );
+    auto vector_halo =
+        Cajita::createHalo( *vector_grid_field, Cajita::NodeHaloPattern<2>() );
+    auto vector_grid_host =
+        Kokkos::create_mirror_view( vector_grid_field->view() );
+
+    // Simple Kokkos::Views may be used to represent particle data.
+    // Create a scalar point field.
+    Kokkos::View<double*, ExecutionSpace> scalar_point_field(
+        Kokkos::ViewAllocateWithoutInitializing( "scalar_point_field" ),
+        num_point );
+    auto scalar_point_host = Kokkos::create_mirror_view( scalar_point_field );
+
+    // Create a vector point field.
+    Kokkos::View<double* [2], ExecutionSpace> vector_point_field(
+        Kokkos::ViewAllocateWithoutInitializing( "vector_point_field" ),
+        num_point );
+    auto vector_point_host = Kokkos::create_mirror_view( vector_point_field );
+
+    // Create a tensor point field.
+    Kokkos::View<double* [2][2], ExecutionSpace> tensor_point_field(
+        Kokkos::ViewAllocateWithoutInitializing( "tensor_point_field" ),
+        num_point );
+    auto tensor_point_host = Kokkos::create_mirror_view( tensor_point_field );
+
+    /***************************************************************************
+     * P2G
+     **************************************************************************/
+    /*
+     * The Cajita::P2G namespace contains several methods for interpolating data
```

Comment:
It would be nice to show this usage as well for one example of each p2g & g2p (particularly based on what Stuart's shown with kernel fusion)